### PR TITLE
feat(rest-client): redesign

### PR DESCRIPTION
**Pull request by michaldudek**
Reimplementation of the rest-client (or bits of it).

## http-client generator

Fixed `-dir` to `-d`.

## client-endpoint generator

Client endpoints are now generated as proper projects in the workspace, so nx can discover them and everything that needs to be done for them to be accessible, is done (e.g. entries into `tsconfig.base.json` etc).

NOTE: Project name for an endpoint is generated from `[directory]-[endpoint-name]-[version]`, so for e.g. `--endpointName users --directory api --endpointVersion 1` the resulting project will be called `api-users-v1`, not just `users` as one might expect. This behavior makes sense to ensure unique names of projects, but may be unexpected to some.

Also, previously uppercase V was being used in folder names, now it will be lowercase `v` (e.g. `v1` instead of `V1`).

Removed default value from `--httpClient` as the default value will always be wrong (it depends on the workspace's `npmScope`). Ideally we could read it dynamically from existing projects, but its impossible, `schema.json` is very static :(

Various fixes to the generated code (proper typing, fixed linting, etc).

Fixed `-dir` to `-d`.

## bump-version generator

That escalated quickly

The generator signature completely changed:

```
nx g @ensono-stacks/rest-client:bump-version users --directory restapi --endpointVersion 2
```

Only name (`users` above) is required. If `--endpointVersion` is omitted, it will just find the highest version and bump by 1.

Each version is generated as a separate project / lib (see above in `client-endpoint` generator).

The generator will also make sure to update all version numbers in the code copied from the previous version).

## Other changes

Moved `normalizeOptions()` function used in multiple packages to `@ensono-stacks/core` for reusability.

Added `copyFiles()` function to `@ensono-stacks/core` as surprisingly `@nrwl/devkit` only supports moving (renaming) files, not copying.

### CODE REVIEW

**Review by nandorszentpeteri — APPROVED** [review]
Nice work!

We have to go through all the plugins now and use the normalize function from the core ;)

File: packages/rest-client/src/generators/bump-version/schema.json [hunk 1]
```diff
@@ -6,21 +6,26 @@
...
+                "index": 0
+            }
         },
-        "endpoint": {
+        "directory": {
             "type": "string",
-            "description": "The endpoint you want to bump to the specified version.",
-            "x-prompt": "What is the endpoint you want to bump to the specified version?"
+            "description": "The directory where the endpoint lib is placed.",
+            "alias": "d"
```
Comment (MetecanAydin) [inline packages/rest-client/src/generators/bump-version/schema.json:21]: Does `d` alias work for you? Seems like they decided against changing this within Nx and kept it to alias dry-run. Recommended is `dir`
https://github.com/nrwl/nx/issues/6571
https://github.com/nrwl/nx/pull/6596

Let me know what you think :)

**Comment by michaldudek** [comment]
> Nice work!
> 
> We have to go through all the plugins now and use the normalize function from the core ;)

I've already done that in this PR. There weren't many, turns out.

File: packages/rest-client/src/generators/bump-version/schema.json (hunk 1 above)
Comment (michaldudek) [reply packages/rest-client/src/generators/bump-version/schema.json]: Indeed! Pushed fix.

**Review by MetecanAydin — APPROVED** [review]

**Review by aizlewood40 — APPROVED** [review]

**Pull request merged**